test/jasmine/tests/transform_sort_test.js

```diff
@@ -273,14 +273,18 @@ describe('Test sort transform interactions:', function() {
         }
 
         function hover(gd, id) {
-            return new Promise(function(resolve) {
+            return new Promise(function(resolve, reject) {
                 gd.once('plotly_hover', function(eventData) {
                     delete gd._lastHoverTime;
                     resolve(eventData);
                 });
 
                 var pos = getPxPos(gd, id);
                 mouseEvent('mousemove', pos[0], pos[1]);
+
+                setTimeout(function() {
+                    reject('plotly_hover did not get called!');
```

Comment:
yeah, that's great! Right, and you don't even need to cancel the timeout because after resolve is called, reject doesn't do anything... nice!

In principle to tell *precisely* what's going on we probably need a message - but that can be added during debugging if we get to that point.